### PR TITLE
refactor(x/ecocredit): update basket validation functions

### PR DESCRIPTION
**Pull request by ryanchristo**
## Description

Ref: #1245

This pull request updates utility functions in the ecocredit module to use a consistent format and check empty string within the validation method rather than individually within each message validation.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/ecocredit/basket/features/msg_create.feature [hunk 1]
```diff
@@ -132,7 +132,7 @@ Feature: MsgCreate
     }
     """
     When the message is validated
-    Then expect the error "name must start with an alphabetic character, and be between 3 and 8 alphanumeric characters long: invalid request"
+    Then expect the error "name: expected format [a-zA-Z][a-zA-Z0-9]{2,7}: parse error: invalid request"
```
Comment (technicallyty) [inline x/ecocredit/basket/features/msg_create.feature:135]: perhaps we should opt for a more human readable error for users who might not understand regex?

File: x/ecocredit/basket/utils.go [hunk 2]
```diff
@@ -41,22 +42,28 @@ func FormatBasketDenom(name, creditTypeAbbrev string, exponent uint32) (string,
...
 	matches := regexBasketName.FindStringSubmatch(name)
 	if matches == nil {
-		return fmt.Errorf("name must start with an alphabetic character, and be between 3 and 8 alphanumeric characters long")
+		return ecocredit.ErrParseFailure.Wrapf("expected format %s", RegexBasketName)
 	}
 	return nil
 }
 
-// ValidateBasketDenom validates a basket denomination conforms to the format
+// ValidateBasketDenom validates a basket denom. The denom conforms to the format
```
Comment (technicallyty) [inline x/ecocredit/basket/utils.go:58]:
```suggestion
// ValidateBasketDenom validates a basket denom. The denom must conform to the format
```

File: x/ecocredit/basket/utils.go [hunk 3]
```diff
@@ -41,22 +42,28 @@ func FormatBasketDenom(name, creditTypeAbbrev string, exponent uint32) (string,
 	return denom, displayDenom, nil
 }
 
-// ValidateBasketName validates a basket denomination conforms to the format
-// described in FormatBasketName. The return is nil if the denom is valid.
+// ValidateBasketName validates a basket name. The name conforms to the format
```
Comment (technicallyty) [inline x/ecocredit/basket/utils.go:45]:
```suggestion
// ValidateBasketName validates a basket name. The name must conform to the format
```

File: x/ecocredit/basket/features/msg_create.feature (hunk 1 above)
Comment (ryanchristo) [reply x/ecocredit/basket/features/msg_create.feature:135]: Yea, regex is probably not the most user friendly. :+1:

Comment (ryanchristo) [reply x/ecocredit/basket/features/msg_create.feature]: https://github.com/regen-network/regen-ledger/pull/1392/commits/6d67b15b6e384a28d25ef959607733704b04d1be

**Review by aleem1314 — APPROVED** [review]

**Pull request merged**